fdbserver/ConfigBroadcaster.actor.cpp

```diff
@@ -253,9 +158,42 @@ class ConfigBroadcasterImpl {
 		return Void();
 	}
 
+	ACTOR static Future<Void> waitForFailure(ConfigBroadcasterImpl* self,
+	                                         Future<Void> watcher,
+	                                         BroadcastClientDetails* client) {
+		wait(success(watcher));
+		self->clients.erase(std::remove(self->clients.begin(), self->clients.end(), *client));
+		return Void();
+	}
+
 public:
-	Future<Void> serve(ConfigBroadcaster* self, ConfigBroadcastFollowerInterface const& cbfi) {
-		return serve(self, this, cbfi);
+	Future<Void> registerWorker(ConfigBroadcaster* self,
+	                            Version lastSeenVersion,
+	                            ConfigClassSet configClassSet,
+	                            Future<Void> watcher,
+	                            ConfigBroadcastInterface broadcastInterface) {
+		actors.add(consumer->consume(*self));
```

Comment:
There should only be one `IConfigConsumer::consume` actor running for the whole `ConfigBroadcaster`. The current implementation creates a new actor for each call to `registerWorker`.